package NYT.NHive.NProto;

import "yt/core/misc/guid.proto";
import "yt/server/hive/hive_manager.proto";

////////////////////////////////////////////////////////////////////////////////

message TReqPing
{
    required NYT.NProto.TGuid src_cell_id = 1;
}

message TRspPing
{
    required int32 last_incoming_message_id = 1;
}

////////////////////////////////////////////////////////////////////////////////

message TReqPostMessages
{
    required NYT.NProto.TGuid src_cell_id = 1;
    required int32 first_message_id = 2;
    repeated TEncapsulatedMessage messages = 3;
}

message TRspPostMessages
{
    required int32 last_acknowledged_message_id = 1;
}

////////////////////////////////////////////////////////////////////////////////

